src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/persistence/PersistenceMapKeyDiagnosticsCollector.java

```diff
@@ -37,81 +37,56 @@ protected String getDiagnosticSource() {
-                    for (PsiAnnotation annotation : allAnnotations) {
-                        String matchedAnnotation = getMatchedJavaElementName(type, annotation.getQualifiedName(),
-                                PersistenceConstants.SET_OF_PERSISTENCE_ANNOTATIONS);
-                        if (matchedAnnotation != null) {
-                            if (PersistenceConstants.MAPKEY.equals(matchedAnnotation))
-                                hasMapKeyAnnotation = true;
-                            else if (PersistenceConstants.MAPKEYCLASS.equals(matchedAnnotation))
-                                hasMapKeyClassAnnotation = true;
-                            else if (PersistenceConstants.MAPKEYJOINCOLUMN.equals(matchedAnnotation)) {
-                                mapKeyJoinCols.add(annotation);
-                            }
-                        }
-                    }
-                    if (hasMapKeyAnnotation && hasMapKeyClassAnnotation) {
-                        // A single field cannot have the same
-                        diagnostics.add(createDiagnostic(field, unit,
-                                Messages.getMessage("MapKeyAnnotationsNotOnSameField"),
-                                PersistenceConstants.DIAGNOSTIC_CODE_INVALID_ANNOTATION, null,
-                                DiagnosticSeverity.Error));
-                    }
-                    if (mapKeyJoinCols.size() > 1) {
-                        validateMapKeyJoinColumnAnnotations(mapKeyJoinCols, field, unit, diagnostics);
-                    }
+                    collectDiagnostics(unit, diagnostics, type, field);
+                }
+            }
+        }
+    }
+
+    private void collectDiagnostics(PsiJavaFile unit, List<Diagnostic> diagnostics,
+                                    PsiClass type, PsiJvmModifiersOwner fieldOrProperty) {
+        List<PsiAnnotation> mapKeyJoinCols = new ArrayList<PsiAnnotation>();
+        boolean hasMapKeyAnnotation = false;
+        boolean hasMapKeyClassAnnotation = false;
+        PsiAnnotation[] allAnnotations = fieldOrProperty.getAnnotations();
+        for (PsiAnnotation annotation : allAnnotations) {
+            String matchedAnnotation = getMatchedJavaElementName(type, annotation.getQualifiedName(),
+                    PersistenceConstants.SET_OF_PERSISTENCE_ANNOTATIONS);
+            if (matchedAnnotation != null) {
+                if (PersistenceConstants.MAPKEY.equals(matchedAnnotation))
+                    hasMapKeyAnnotation = true;
+                else if (PersistenceConstants.MAPKEYCLASS.equals(matchedAnnotation))
+                    hasMapKeyClassAnnotation = true;
+                else if (PersistenceConstants.MAPKEYJOINCOLUMN.equals(matchedAnnotation)) {
+                    mapKeyJoinCols.add(annotation);
                 }
             }
         }
+        if (hasMapKeyAnnotation && hasMapKeyClassAnnotation) {
+            // A single field or property cannot have the same
```

Comment:
I know this comment did not originate with this PR, but I'm not clear on what it means.

Reply:
It's funny. I read this too and wanted to improve the comment, but also couldn't make sense of what the original author meant. I think this just came along for the ride from the port from LSP4Jakarta so would expect to see this comment in that code base too. I opted to just leave it alone.